Cactro Rapido one-way linking

MowFord opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Closes #3342 by creating a new mobMod that designates that the mob links in one direction. Mob linking is handled by the mob's party, built at zone init. It adds all mobs with linking enabled into a party of all other mobs in the zone with the same familyid. Sabotenders were not set to link, so they were not in the party. 

The new mod makes mobs with the mod not execute `TryLink` on its own party members. Therefore, it makes Cactrot function exactly the way it's supposed to: It links when passing by a player engaged with a cactuar, but when in combat, it doesn't attract other mobs of the cactuar family to join the fight.

Note that WingsXI and ASB both use a different (new) mobMod called `ATTRACT_FAMILY_NM` but both implementations do a second loop of every mob in the zone to test linking, when 2-way linking already has an established method of just making the mobs join the same party. Also, it's only used for this one NM.

I believe this PR is a more generalized solution. And can be set up to do exactly what `ATTRACT_FAMILY_NM` does by simply setting the non-nm `mob_pool` to linking, then set `mobMod.NO_LINK` to zero.

Also sidenote: https://ffxiclopedia.fandom.com/wiki/Cactrot_Rapido shows a picture of Cactro without dust around him, so I'm assuming his animation sub should change while engaged.

<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->

## Steps to test these changes

[Change in action](https://i.imgur.com/RtQsphs.mp4)

 - Cactrot was engaged by having him link with another Sabotender (this behavior is already coded to set his roamflag to ignore anything but linking). 
 - I then killed the sabotender and pulled Cactro to another Sabotender; no linking happened. 
 - I then kept him aggroed, pulled that Sabotender to a third Sabotender. No linking happened until the 2nd Sabotender was in range, and they linked together.

This confirms the change has no change to traditional linking, and the new behavior behaves properly in the one direction.

Commit:
add mobmod for one-way linking

MowFord commented:
According to this https://ffxiclopedia.fandom.com/wiki/Sabotender they shouldn't link, which actually makes the meat of what I implemented pointless (for this particular NM). 

Perhaps a worthwhile mobmod addition? If not I can close this and make a new PR that solves the Cactrot problem by simply
 - setting Sabotender mob_pool to linking in the db
 - setting `Sabotender.lua` in East Altepa to add the mobMod `NO_LINK`

Also note I've confirmed previously that the `mob_pool` for East Altepa Sabotenders is only used in that zone

TeoTwawki commented:
Honestly kinda weirded out that for once someone is trying this the right way core side with a re-usable method, and not a performance chewing hack.

And iirc sabo's should agro but not link so ty for catching that, was probably done in an attempt to make the NM work.

MowFord commented:
I got all in my head thinking about this. So like I said we can leave the new mob mod as it may come in handy somewhere else, but for Cactrot we don't need it. Just as simple as setting Sabotenders to link, but undoing that when they spawn.

TeoTwawki commented:
> Commit title is too long.
Maximum length is 72 characters.
If you need to provide a more detailed summary of your changes you should provide a multi-line commit message

if you do a rebase you can edit your messages where that ellipses got placed is the cut off point
> enable Sabotender linking in east altepa for cactro to be in same mob…
… party, then give cactro one-way linking mobmod

![image](https://user-images.githubusercontent.com/6871475/232921834-d03108f8-95bc-4cb6-9bc4-6860fe399901.png)

TeoTwawki commented:
> I got all in my head thinking about this. So like I said we can leave the new mob mod as it may come in handy somewhere else, but for Cactrot we don't need it. Just as simple as setting Sabotenders to link, but undoing that when they spawn.

I'd like to have the mobmod anyway, and I think its fine to use it here. its more obvious to people what's going on then. I think at one point we did not have the ability to toggle it after spawn, and this was never revisited after (I wasn't even aware we could, rapid has been busted pretty since day 1 - even from the day he was added to dsp). If we do it by changing the link state, makes more sense to change Rapido's than all of theirs doesn't it?

TeoTwawki commented:
oh I see, `NO_LINK`. sorry posting while very tired. I have TeoMod(NO_SLEEP) on me today

Commit:
enable Sabotender linking in east altepa
This puts cactrot in same mob party, which allows linking
Finally give cactro one-way linking mobmod

Commit:
Disable sabotender linking altogether

MowFord commented:
Just curious what the hold-up is. No rush, just want to make sure it's not waiting on me to do something.

TeoTwawki commented:
> Just curious what the hold-up is. No rush, just want to make sure it's not waiting on me to do something.

Reviewers just needing some time off their dayjobs + some touching grass time is all. I'm at work right now in fact and after work I have a crapton of house chores to catch up on.

Xaver-DaRed reviewed (approved):
Simple, clean and well commented. Nice job

TeoTwawki reviewed (approved):
thought I wasn't gonna be back here till sunday, decided to slack on my chores.

TeoTwawki commented on sql/mob_pools.sql:
```diff
@@ -3478,7 +3478,7 @@ INSERT INTO `mob_pools` VALUES (3422,'Ruphuabo','Ruphuabo',194,0x000083040000000
 INSERT INTO `mob_pools` VALUES (3424,'Sabertooth_Tiger','Sabertooth_Tiger',242,0x0000340100000000000000000000000000000000,1,1,7,240,100,0,1,0,0,0,0,0,238,131,0,0,0,0,0,242,242);
 INSERT INTO `mob_pools` VALUES (3425,'Sable-tongued_Gonberry','Sable-tongued_Gonberry',243,0x0000C60400000000000000000000000000000000,4,4,6,240,100,0,1,0,1,18,0,32,167,157,0,0,2,0,0,243,243);
-INSERT INTO `mob_pools` VALUES (3426,'Sabotender','Sabotender',212,0x0000740100000000000000000000000000000000,2,2,7,360,100,0,1,0,0,0,0,0,6461,131,0,0,0,0,0,212,212);
+INSERT INTO `mob_pools` VALUES (3426,'Sabotender','Sabotender',212,0x0000740100000000000000000000000000000000,2,2,7,360,100,0,1,0,1,0,0,0,6461,131,0,0,0,0,0,212,212);
```
pr doesn't hinge on it but could I get comment at the end of this row telling ppl that the linking is adjusted in script? Just so they don't undo your work here thinking its wrong.

MowFord replied:
Done.

Commit:
Add info to justfiy setting a mob to link
That shouldn't link

MowFord commented:
> Reviewers just needing some time off their dayjobs

Fair enough.

I've updated the requested comment and run dbtool to ensure the `.sql` file still imports properly.
